File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int sum = 0 , count = 0 ; 
        unordered_map<int , int > psm ; 
        psm[0] = 1 ;
        for(int i = 0 ; i < nums.size() ; i++)
        {
            sum += nums[i]; 
            int rem = sum - k ; 
            if(psm.find(rem) != psm.end())
            {
                count += psm[rem]; 
            }
            psm[sum]++;
        }
        return count ; 
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        res = 0 
        currsum = 0 
        psm = { 0 : 1}
        for n in nums:
            currsum += n 
            diff = currsum - k

            res += psm.get(diff,0)
            psm[currsum] = 1 + psm.get(currsum,0) 

        return res    